handler.py

```diff
@@ -14,38 +16,120 @@
 SQS_CLIENT = boto3.client('sqs')
 
 
-def runScheduledObservatoryScan(event, context):
+def addPortScanToQueue(event, context):
+    data = json.loads(event['body'])
+    target = Target(data.get('target'))
+
+    if target.isValid():
+        print(SQS_CLIENT.send_message(
+              QueueUrl=os.getenv('SQS_URL'),
+              MessageBody="portscan|" + target.targetname
+              ))
+
+        return Response({
+            "statusCode": 200,
+            "body": json.dumps({'OK': 'target added to the queue'})
+        }).with_security_headers()
+
+    else:
+        logger.error("Target Validation Failed of: " +
+                     json.dumps(target.targetname))
+        return Response({
+             "statusCode": 400,
+             "body": json.dumps({'error': 'target was not valid or missing'})
+         }).with_security_headers()
 
+
+def runScheduledPortScan(event, context):
```

Comment:
Should we consider moving the queue and run methods for each scanner outside of the handler logic?

I only say that because it will likely make it  easier to mock and  keep  the handler code as simplistic as possible, which I think conventionally makes the most sense.

Reply:
I think this is probably better for for another branch, it's a fair bit of re-write of the code.
@claudijd you can give it a go if you want.

Reply:
ok, once we land